gdnative-core/src/macros.rs

```diff
@@ -323,14 +324,11 @@ macro_rules! godot_wrap_constructor {
             this: *mut $crate::sys::godot_object,
             _method_data: *mut $crate::libc::c_void,
         ) -> *mut $crate::libc::c_void {
-            use std::boxed::Box;
-            use std::cell::RefCell;
-
             // let val = $c($crate::NativeInstanceHeader{ this: this });
             let val = $c();
 
-            let wrapper = Box::new(RefCell::new(val));
-            Box::into_raw(wrapper) as *mut _
+            let wrapper = <$_name as $crate::NativeClass>::UserData::new(val);
+            Rc::into_raw(wrapper) as *mut _
```

Comment:
Shouldn't this be configurable as well? Is this a remnant of the initial implementation without the user customization?

Reply:
Indeed it is, thanks for catching this! The macro is not used in tests so it went through unnoticed.